sklearn/experimental/_column_transformer.py

```diff
@@ -0,0 +1,220 @@
+            hstack of results of transformers. sum_n_components is the
+            sum of n_components (output dimension) over transformers.
+        """
+        try:
+            return super(ColumnTransformer, self).fit_transform(X, y=y,
+                                                                **fit_params)
+        except ValueError as e:
+            if "Got X with X.ndim=1. Reshape your data" in str(e):
+                raise ValueError(_ERR_MSG_1DCOLUMN)
+            else:
+                raise e
+
+    def transform(self, X):
+        """Transform X separately by each transformer, concatenate results.
+
+        Parameters
+        ----------
+        X : array-like or DataFrame of shape [n_samples, n_features]
+            Input data, of which specified subsets are used to fit the
+            transformers.
+
+        Returns
+        -------
+        X_t : array-like or sparse matrix, shape (n_samples, sum_n_components)
+            hstack of results of transformers. sum_n_components is the
+            sum of n_components (output dimension) over transformers.
+        """
+        try:
+            return super(ColumnTransformer, self).transform(X)
+        except ValueError as e:
+            if "Got X with X.ndim=1. Reshape your data" in str(e):
+                raise ValueError(_ERR_MSG_1DCOLUMN)
+            else:
+                raise e
+
+
+def _get_column(X, key):
+    """
+    Get feature column(s) from input data X.
+
+    Supported input types (X): numpy arrays, sparse arrays and dataframes
+
+    Supported key types (key):
+    - scalar: output is 1D
+    - lists, slices, boolean masks: output is 2D
+
+    Supported key data types:
+
+    - integer or boolean mask (positional):
+        - supported for (sparse) arrays or dataframes
+    - string (key-based):
+        - only supported for dataframes
+        - So no keys other than strings are allowed (while in principle you
+          can use any hashable object as key).
+
+    """
+    if X is None:
+        return X
+
+    # check whether we have string column names or integers
+    if (isinstance(key, int)
+            or (isinstance(key, list)
+                and all(isinstance(col, int) for col in key))
+            or (isinstance(key, slice)
+                and isinstance(key.start, (int, type(None)))
+                and isinstance(key.stop, (int, type(None))))):
+        column_names = False
+    elif (isinstance(key, six.string_types)
+            or (isinstance(key, list)
+                and all(isinstance(col, six.string_types) for col in key))
+            or (isinstance(key, slice)
+                and isinstance(key.start, (six.string_types, type(None)))
+                and isinstance(key.stop, (six.string_types, type(None))))):
+        column_names = True
+    elif hasattr(key, 'dtype') and np.issubdtype(key.dtype, np.bool):
+        # boolean mask
+        column_names = False
+        if hasattr(X, 'loc'):
```

Comment:
yep, they work with loc, but apparantly not with iloc (and for this reason I had to special case it ...). Also learned this just now :-) 

(Intending to open an issue about that on pandas, can't remember the reason to do this)